### PR TITLE
[JSON RPC] expose mempool status codes

### PR DESCRIPTION
**Pull request by christinacelee**
## Summary

Expose mempool error status codes as custom Libra server codes

### CODE REVIEW

File: json-rpc/src/errors.rs [hunk 1]
```diff
@@ -10,14 +14,21 @@ use serde_json::Value;
...
     VmVerificationError = -32002,
     VmInvariantViolationError = -32003,
     VmDeserializationError = -32004,
     VmExecutionError = -32005,
     VmUnknownError = -32006,
-    // TODO expose Mempool insertion status as server code
+
+    // Mempool errors - see `MempoolStatusCode` for specs
+    MempoolInvalidSeqNumber = -32007,
+    MempoolIsFull = 32008,
```
Comment (davidiw) [inline json-rpc/src/errors.rs]: why isn't this one negative?

Are there any tests to ensure that these values make sense?

Comment (christinacelee) [reply json-rpc/src/errors.rs]: ah that's a really good catch, thanks!

re: testing: Rust doesn't support iterating through enums, even static ones, so to test the enum values we'd either need to use crates like [this](https://docs.rs/strum/0.18.0/strum/ ), or keep a static list of the enum variants to iterate through.. neither of these options seem ideal :/

Comment (davidiw) [reply json-rpc/src/errors.rs]: Well, Rust might not, but there's always a macro for that :P

https://docs.rs/enum-iterator/0.6.0/enum_iterator/derive.IntoEnumIterator.html

**Pull request closed without merging**